### ISSUE TITLE
Layouts can/may use old window reserved area value

### ISSUE THREAD

**Issue by zakk4223** [hyprwm/Hyprland#5818]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 2e763764bfdaee2039961c6c220b5d6af31712d4  (pointer-constraints: don't warp on entering a confined pointer).
Date: Sat Apr 27 16:15:33 2024
Tag: v0.39.1-91-g2e763764, commits: 4551

flags: (if any)


System Information:
System name: Linux
Node name: blkheart
Release: 6.8.7-zen1-2-zen
Version: #1 ZEN SMP PREEMPT_DYNAMIC Fri, 19 Apr 2024 09:51:33 +0000


GPU information: 
0d:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Navi 21 [Radeon RX 6800/6800 XT / 6900 XT] [1002:73bf] (rev c0) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:


```

</details>

### Bug or Regression?

Bug

### Description

If a window decorator dynamically changes the size of the decoration and calls repositionDeco() the layout won't be aware of the change until the next time something triggers a layout/window recalculation.

### How to reproduce

I don't think there's an existing deco that triggers this: I discovered it while adding vertical tabs to the GroupBar deco.

When the active/visible window changes in the group, the sequence is:
layout receives replaceWindowDataWith, which calls
applyNodeDataToWindow, which asks the window for its reserved area.
It will get the OLD value of the reserved area for that window, which is whatever the reserved area was when the window was first added to the group.
The layout then sizes the window with the wrong reserved area.
The decos draw() call is eventually called and then it notices the difference in assigned geometry vs what it wants and calls repositionDeco(). 
This triggers the reserved area recalculation, but there's no trigger to tell the layout to readjust.

I can "fix" it by calling recalculateWindow in onPositioningReply, but this feels like a more general problem that decos shouldn't have to deal with individually.

Can the deco positioner just call recalculateWindow when it changes the reserved area, or is there a better/more efficient way?

**Comment by vaxerski** [comment]
deco positioner should do it probably yea

**Comment by zakk4223** [comment]
fixed in d2899a6c272d4619295543b5b4c77b40fa1934bf
